ui/src/masking.ts

```diff
@@ -0,0 +1,76 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import * as escape from 'escape-string-regexp';
+import * as os from 'os';
+import { IActionContext, IParsedError } from "../index";
+import { parseError } from "./parseError";
+import { AzExtTreeItem } from './tree/AzExtTreeItem';
+
+let _extValuesToMask: string[] | undefined;
+function getExtValuesToMask(): string[] {
+    if (!_extValuesToMask) {
+        try {
+            _extValuesToMask = [os.userInfo().username];
+        } catch {
+            _extValuesToMask = [];
+        }
+    }
+    return _extValuesToMask;
+}
+
+export function addExtensionValueToMask(...values: string[]): void {
+    const extValuesToMask: string[] = getExtValuesToMask();
+    for (const v of values) {
+        if (!extValuesToMask.includes(v)) {
+            extValuesToMask.push(v);
+        }
+    }
+}
+
+/**
+ * Example id: /subscriptions/00000000-0000-0000-0000-00000000/resourceGroups/rg1/providers/Microsoft.Web/sites/site1
+ */
+export function addValuesToMaskFromAzureId(context: IActionContext, node: AzExtTreeItem): void {
+    const parts: string[] = node.fullId.toLowerCase().split('/');
+    if (parts[1] === 'subscriptions' && parts[3] === 'resourcegroups') { // NOTE: subscription id is already added to extValuesToMask elsewhere
+        context.valuesToMask.push(parts[4]);
+
+        if (parts[5] === 'providers' && parts[6]?.startsWith('microsoft.') && parts[8]) {
```

Comment:
Would it be possible to search the array for 'providers' index and then get the value of the index + 1?  I'm pretty sure with resource ids, that they just keep tacking on parts to the end, but not 100% confident that'll always be the case

Reply:
What problem are you trying to solve? Can you give an example id maybe?

Reply:
Never mind, I thought that it'd be possible that an id didn't follow this exact formatting.  But according to this: https://docs.microsoft.com/en-us/rest/api/resources/resources/getbyid that's always the expected structure.